sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveSessionCatalog.scala

```diff
@@ -174,6 +175,18 @@ private[sql] class HiveSessionCatalog(
   // xpath_double, xpath_float, xpath_int, xpath_long, xpath_number,
   // xpath_short, and xpath_string.
   override def lookupFunction(name: FunctionIdentifier, children: Seq[Expression]): Expression = {
+    try {
+      subLookupFunction(name, children)
+    } catch {
```

Comment:
@dongjoon-hyun the current fix is quite brittle; this will fail again as soon as we pass in an argument with a slightly different value. The Analyzer will create casts to the proper type if we implement `ExpectsInputTypes`. So this seems like the best course of action. It might not be the easiest fix, or entirely possible; but I'd prefer to try this first.